test/org/sagebionetworks/bridge/dynamodb/DynamoHealthDataDaoDdbTest.java

```diff
@@ -86,4 +99,62 @@ public void ddbRoundtrip() throws Exception {
             mapper.delete(savedRecord);
         }
     }
+
+    @Test
+    public void getRecordsByHealthCodeCreatedOnSchemaId() throws IOException {
+        // store a test healthdata record in ddb
+        DynamoHealthDataRecord testRecord = new DynamoHealthDataRecord();
+        testRecord.setCreatedOn(CREATED_ON);
+        testRecord.setHealthCode(HEALTH_CODE);
+        testRecord.setSchemaId(SCHEMA_ID);
+        testRecord.setId(RECORD_ID);
+        testRecord.setSchemaRevision(SCHEMA_REV);
+        testRecord.setUploadDate(UPLOAD_DATE);
+        testRecord.setUploadedOn(UPLOADED_ON);
+        testRecord.setUploadId(UPLOAD_ID);
+        testRecord.setUserSharingScope(ParticipantOption.SharingScope.SPONSORS_AND_PARTNERS);
+        testRecord.setUserExternalId(USER_EXTERNAL_ID);
+        testRecord.setUserDataGroups(TestConstants.USER_DATA_GROUPS);
+        testRecord.setData(BridgeObjectMapper.get().readTree(DATA_TEXT));
+        testRecord.setMetadata(BridgeObjectMapper.get().readTree(METADATA_TEXT));
+        testRecord.setStudyId(TestConstants.TEST_STUDY_IDENTIFIER);
+
+        mapper.save(testRecord);
```

Comment:
It might be worthwhile to move record creation to an @Before and record deleting to an @After. This removes duplicate code, and if the test fails and then the record deletion also fails, we get a more useful error message.